src/Lean/Elab/MutualDef.lean

```diff
@@ -708,7 +708,8 @@ private def levelMVarToParamHeaders (views : Array DefView) (headers : Array Def
     let mut newHeaders := #[]
     for view in views, header in headers do
       if view.kind.isTheorem then
-        newHeaders := newHeaders.push { header with type := (← levelMVarToParam' header.type) }
+        newHeaders := newHeaders.push { header with type :=
+          (← monadMap (@withLevelNames (levelNames := header.levelNames)) (levelMVarToParam' header.type)) }
```

Comment:
Not entirely sure why the implicit lambda didn't work

Reply:
It is worth investigating, but let's avoid `monadMap` here. We should use the idiom we have in the rest of the code.
```lean
def withLevelNamesImp (levelNames : List Name) (x : TermElabM α) : TermElabM α := do
...

def withLevelNames [MonadFunctorT TermElabM m] [Monad m] (levelNames : List Name) (x : m α) : m α := 
...
```

Reply:
Okay, I had to restructure this part anyway to fix the last reported bug, which was due to not storing the new level names between independent invocations of `levelMVarToParam`. Fixing this removed the need for the state transformer, simplifying the code.